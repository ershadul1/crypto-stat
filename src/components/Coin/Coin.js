import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Coin.module.css';
import roundNum from '../../helpers/roundNum';

const Coin = ({ data, currency }) => (
  <div>
    <Link className={styles.container} to={`/details/${data.id}`}>
      <div>
        <img className={styles.logo} src={data.image} alt={`${data.name}-logo`} />
        {data.name}
      </div>
      <div>
        {`${currency.toUpperCase()} ${new Intl.NumberFormat('en-IN', { maximumSignificantDigits: 3 }).format(data.current_price)}`}
      </div>
      <div>
        {`${currency.toUpperCase()} ${new Intl.NumberFormat('en-IN', { maximumSignificantDigits: 3 }).format(data.market_cap)}`}
      </div>
      <div>
        {roundNum(data.price_change_percentage_24h)}
        %
      </div>
      <div>
        {roundNum(data.price_change_percentage_7d_in_currency)}
        %
      </div>
    </Link>
  </div>
);

Coin.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    current_price: PropTypes.number.isRequired,
    market_cap: PropTypes.number.isRequired,
    price_change_percentage_24h: PropTypes.number.isRequired,
    price_change_percentage_7d_in_currency: PropTypes.number.isRequired,
  }).isRequired,
  currency: PropTypes.string.isRequired,
};

export default Coin;
