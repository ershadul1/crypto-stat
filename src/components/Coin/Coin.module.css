@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');

.container {
  display: flex;
  justify-content: space-around;
  align-items: center;
  text-decoration: none;
  padding: 20px 0;
  border-bottom: 1px solid rgb(212, 211, 211);
}

.container div {
  width: 200px;
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  text-align: right;
  color: #000;
}

.cell {
  display: flex;
  align-items: center;
}

.logo {
  width: 30px;
  height: 30px;
  display: block;
  margin-right: 10px;
}

.container:hover {
  background-color: rgb(246, 246, 246);
}
