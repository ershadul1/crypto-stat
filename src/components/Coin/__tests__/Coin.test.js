import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Coin from '../Coin';

it('Coin snapshot test', () => {
  const data = {
    id: 'bitcoin',
    name: 'Bitcoin',
    image: 'https://example.com/image.jpg',
    current_price: 20,
    market_cap: 500,
    total_volume: 100,
    price_change_percentage_24h: 5,
    price_change_percentage_7d_in_currency: 10,
  };
  const currency = 'usd';
  const tree = renderer
    .create(
      <BrowserRouter>
        <Coin data={data} currency={currency} />
      </BrowserRouter>,

    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
