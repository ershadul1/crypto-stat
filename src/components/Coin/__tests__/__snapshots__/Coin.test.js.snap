// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin snapshot test 1`] = `
<div>
  <a
    className="container"
    href="/details/bitcoin"
    onClick={[Function]}
  >
    <div
      className="cell"
    >
      <img
        alt="Bitcoin-logo"
        className="logo"
        src="https://example.com/image.jpg"
      />
      <p>
        Bitcoin
      </p>
    </div>
    <div>
      USD 20
    </div>
    <div>
      USD 500
    </div>
    <div>
      USD 100
    </div>
    <div
      id="green"
    >
      5
      %
    </div>
    <div
      id="green"
    >
      10
      %
    </div>
  </a>
</div>
`;
